### PR TITLE
FEATURE: QA-9438 Adding some charts at Visualization Component of Live Examples for testing

### PR DESCRIPTION
**Pull request by mluu3**

### CODE REVIEW

**Comment by mluu3** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-876.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3750/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3001/ : FAILURE in 1m 54s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3751/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/993/ : FAILURE in 29m 29s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3752/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3002/ : FAILURE in 1m 22s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3754/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3004/ : SUCCESS in 2m 19s

**Comment by mluu3** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-876.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3755/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/994/ : SUCCESS in 24m 48s

File: examples/test/VisualizationByIdentifier_test.js [hunk 1]
```diff
@@ -35,3 +44,44 @@ test('Table visualization should render', async (t) => {
...
+
+test('Pie chart should render', async (t) => {
+    await checkRenderChart('.s-visualization-pie', t);
+});
+
+test('Donut chart should render', async (t) => {
+    await checkRenderChart('.s-visualization-donut', t);
+});
+
+
```
Comment (xMort) [inline examples/test/VisualizationByIdentifier_test.js]: Please remove the second blank row.

File: examples/test/VisualizationByUri_test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+// (C) 2007-2018 GoodData Corporation
+import { Selector } from 'testcafe';
+import { config } from './utils/config';
+import { loginUsingGreyPages } from './utils/helpers';
+
+fixture('Visualization') // eslint-disable-line no-undef
+    .page(config.url)
+    .beforeEach(loginUsingGreyPages(`${config.url}/visualization/visualization-by-uri`));
+
+async function checkRenderChart(selector, t) {
```
Comment (xMort) [inline examples/test/VisualizationByUri_test.js]: Wouldn't `checkRenderedChart` be better as a method name?

File: examples/src/routes/_list.js [hunk 3]
```diff
@@ -39,17 +40,23 @@ export const advancedUseCasesRoutes = [
     { path: '/advanced/chart-configuration', title: 'Chart Configuration', Component: ChartConfiguration }
 ];
 
+export const visualizationUseCasesRoutes = [
+    { path: '/visualization/visualization-by-uri', title: 'Visualization By Uri', Component: VisualizationByUri },
+    { path: '/visualization/visualization-by-identifier', title: 'Visualization By Identifier', Component: VisualizationByIdentifier }
```
Comment (xMort) [inline examples/src/routes/_list.js]: Just a small note: I would name the titles as `Visualization by URI` and `Visualization by identifier` (note the letter cases).

File: examples/src/routes/VisualizationByUri.jsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import VisualizationPieExampleSRC from '!raw-loader!../components/VisualizationPieByUriExample'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved, import/extensions, import/first
+import VisualizationDonutExampleSRC from '!raw-loader!../components/VisualizationDonutByUriExample'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved, import/extensions, import/first
+import VisualizationTreemapExampleSRC from '!raw-loader!../components/VisualizationTreemapByUriExample'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved, import/extensions, import/first
+import VisualizationHeatmapExampleSRC from '!raw-loader!../components/VisualizationHeatmapByUriExample'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved, import/extensions, import/first
+
+export const VisualizationByUri = () => (
+    <div>
+        <h1>Visualization</h1>
+
+        <p>These are examples of the generic Visualization component.</p>
```
Comment (xMort) [inline examples/src/routes/VisualizationByUri.jsx]: I would specify that this test is about URI.
`These are the examples of the generic Visualization component that uses URI to identify insights.`

File: examples/src/routes/VisualizationByIdentifier.jsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+import VisualizationPieExampleSRC from '!raw-loader!../components/VisualizationPieByIdentifierExample'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved, import/extensions, import/first
+import VisualizationDonutExampleSRC from '!raw-loader!../components/VisualizationDonutByIdentifierExample'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved, import/extensions, import/first
+import VisualizationTreemapExampleSRC from '!raw-loader!../components/VisualizationTreemapByIdentifierExample'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved, import/extensions, import/first
+import VisualizationHeatmapExampleSRC from '!raw-loader!../components/VisualizationHeatmapByIdentifierExample'; // eslint-disable-line import/no-webpack-loader-syntax, import/no-unresolved, import/extensions, import/first
+
+export const VisualizationByIdentifier = () => (
+    <div>
+        <h1>Visualization</h1>
+
+        <p>These are examples of the generic Visualization component.</p>
```
Comment (xMort) [inline examples/src/routes/VisualizationByIdentifier.jsx]: I would specify that this test is about URI.
`These are the examples of the generic Visualization component that uses URI to identify insights.`

File: examples/src/routes/_list.js (hunk 3 above)
Comment (xMort) [reply examples/src/routes/_list.js]: Maybe you can use these titles in <h1> titles in the "pages" themselves.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3761/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3011/ : SUCCESS in 4m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3764/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3015/ : SUCCESS in 6m 46s

**Comment by mluu3** [comment]
Hi @tomasreichmann , please help me review and approve for this PR

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3769/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3018/ : SUCCESS in 2m 32s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/818/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3019/ : SUCCESS in 2m 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1609/ : SUCCESS in 12m 28s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/817/ : SUCCESS in 20m 31s

**Pull request merged**